docs/internals/controls-package.md

```diff
@@ -9,15 +9,275 @@ description: "The Control Package implements feedback control on an endpoint."
+
+```go
+type Controllable interface {
+  SetState(ctx context.Context, state []*Signal) error
+  State(ctx context.Context) ([]float64, error)
+}
+```
+
+### Options
+
+`Options` is a struct that contains all of the optional parameters that can be used to customize a control loop during the setup. Since they are all optional, the only options that need to be set are those that you wish to change from the default.
+
+<!-- prettier-ignore -->
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| `PositionControlUsingTrapz` | bool | Adds a trapezoidalVelocityProfile block to the control config to allow for position control of a component. <br> Default: false |
+| `SensorFeedback2DVelocityControl` | bool | Adds linear and angular blocks to a control config in order to use the sensorcontrolled base component for velocity control. <br> Default: false |
+| `DerivativeType` | string | The type of derivative to be used for the derivative block of a control config. <br> Default: `"backward1st1"` |
+| `UseCustomConfig` | bool | Used if the necessary config cannot be created using the control loop setup functions. <br> Default: false |
+| `CompleteCustomConfig` | control.Config | The custom control config to be used instead of the config created by the control loop setup functions. <br> Default: control.Config{} |
+| `NeedsAutoTuning` | bool | True when the loop needs to be auto-tuned. This will be set to true automatically if all PID values are 0. <br> Default: false |
+| `LoopFrequency` | float64 | The frequency at which the control loop should run. <br> Default: 50 Hz |
+| `ControllableType` | string | The type of component the control loop will be set up for, currently a base or motor. <br> Default: `"motor_name"` |
+
+```go
+type Options struct {
+  PositionControlUsingTrapz bool
+  SensorFeedback2DVelocityControl bool
+  DerivativeType string
+  UseCustomConfig bool
+  CompleteCustomConfig Config
+  NeedsAutoTuning bool
+  LoopFrequency float64
+  ControllableType string
+}
+```
+
+The built in control loop setup is only structured to work with an encoded motor or a sensor controlled base. If you wish to use a different setup, you can utilize the options by setting `UseCustomConfig` to `true` and `CompleteCustomConfig` to your custom control loop config of type `control.Config`. The [Control Blocks](/internals/controls-package/#controlblocks) section details the different options for control blocks and how to create a `control.Config`.
+
+### BlockConfig
+
+`BlockConfig` is struct of the configuration for an individual control block. You will have to build individual BlockConfigs if you utilize the `UseCustomConfig` option. Each block type requires different attributes, which are outlined in the [Control Blocks](/internals/controls-package/#controlblocks) section.
```

Comment:
this and the link on line 265 also aren't passing checks... but I think that probably makes sense because there's a space in the `Control Blocks` title... how do I account for that in the hash?